### DISCUSSION TITLE
K3S for IoT remote device

### DISCUSSION THREAD

**Discussion by solinter** [k3s-io/k3s#7735] [General]
Trying to test a POC with a scenarios of K3S Service node running in Cloud and one Agent node running remotelly.

Supposing that one deployment contains a node-selector by this unique remote node and the architecture behold the case of disconnection between the Server and Agent, in any case the POD assigned to remote Agent should be running.

Tested cases:

1. Server node goes down permanently, POD still running in remote Agent. OK
2. Server node goes down permanently and remote Agent is being restarted. When the remote Agent is starting up, the Server still down and previously scheduled POD is not started up until the Server become Up. KO

How to configure the agent correctly in order to recover all previously scheduled PODs after the restart and when the Server still down?

**Comment by brandond** [comment]
Agents cannot start without a server. This is a basic requirement.

**Reply by solinter** [reply]
Hm, but for IoT it would be a requirement to have a reliable "Embedded" system. But in this case it will depends on connection with Server after restart.

**Reply by brandond** [reply]
Yep. Kubernetes isn't designed to be used in a detached deployment model. You need a reliable control plane.
